ddprofiler/app/src/main/java/ddprofiler/preanalysis/PreAnalyzer.java

```diff
@@ -130,26 +150,112 @@ public Map<Attribute, Values> readRows(int num) {
 
     private void calculateDataTypes(Map<Attribute, List<String>> data) {
         // estimate data type for each attribute
-        for (Entry<Attribute, List<String>> e : data.entrySet()) {
-            Attribute a = e.getKey();
-            // Only if the type is not already known
-            if (!a.getColumnType().equals(AttributeType.UNKNOWN))
+        for (Entry<Attribute, List<String>> entry : data.entrySet()) {
+            Attribute attribute = entry.getKey();
+
+            // If the type is known, skip
+            if (!attribute.getColumnType().equals(AttributeType.UNKNOWN))
                 continue;
-            AttributeType aType;
+
+            AttributeType attributeType;
             if (pc.getBoolean(ProfilerConfig.EXPERIMENTAL)) {
                 // In experimental mode - force all data to be strings
-                aType = AttributeType.STRING;
+                attributeType = AttributeType.STRING;
             } else {
-                aType = typeOfValue(e.getValue());
+                attributeType = typeOfValue(entry.getValue());
```

Comment:
this function naming is kinda confusing right now since we have two kinds of data types - basic data types and semantic types. It would be better if you can come up with a clearer name

Reply:
Yes, it is quite confusing now. In [1601d81](https://github.com/TheDataStation/ver/pull/71/commits/1601d81fb1827ba521ca82de3697c546cafc306b), I renamed this line into `attributeDataType`. I thought about "primitiveAttributeType", but it wouldn't be correct, since String is not primitive. 

Maybe we can also rename `AttributeType` into `AttributeDataType`, and the variable `columnType` into `columnDataType`, but this would make changes in lots of files, which is scary. What do you think?

Reply:
maybe rename function `typeofvalue` to `basicTypeOfValue`? by `basic`, I mean types that are predefined in the programming language.

Reply:
I have renamed `typeOfValue` to `dataTypeOfValue` at 93a4a65ef69a0b1bbb0e60fd82d6a7ac98c300bb. I think it make sense and we can differentiate a column's type to data type and semantic type.